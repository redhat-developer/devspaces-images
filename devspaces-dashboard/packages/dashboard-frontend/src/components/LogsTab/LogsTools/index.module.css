.logsTools {
  height: 20px;
  margin: 10px 10px 5px;
}

.logsTools > * {
  float: right;
  height: inherit;
  font-size: small;
  user-select: none;
}

.logsTools a {
  text-decoration: none;
}

.logsTools button {
  padding: 0;
  font-size: inherit;
  outline: none;
}

.logsTools svg {
  margin-right: 8px;
}
