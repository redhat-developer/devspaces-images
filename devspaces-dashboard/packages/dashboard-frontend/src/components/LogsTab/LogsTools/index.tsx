/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import { Flex, FlexItem, Button, Divider } from '@patternfly/react-core';
import { CompressIcon, DownloadIcon, ExpandIcon } from '@patternfly/react-icons/dist/js/icons';
import { getBlobUrl } from '../../../services/helpers/tools';

import styles from './index.module.css';

type Props = {
  logs: string[] | undefined;
  preventPostMessage: boolean;
  handleExpand: (isExpand: boolean) => void;
};

type State = {
  isExpanded: boolean;
};

class LogsTools extends React.PureComponent<Props, State> {
  private readonly handleExpand: () => void;

  constructor(props: Props) {
    super(props);

    this.state = {
      isExpanded: false,
    };

    this.handleExpand = () => {
      if (this.state.isExpanded) {
        if (this.props.preventPostMessage === false) {
          window.postMessage('show-navbar', '*');
        }
        const isExpanded = false;
        this.setState({ isExpanded });
        this.props.handleExpand(isExpanded);
      } else {
        if (this.props.preventPostMessage === false) {
          window.postMessage('hide-navbar', '*');
        }
        const isExpanded = true;
        this.setState({ isExpanded });
        this.props.handleExpand(isExpanded);
      }
    };
  }

  public render(): React.ReactElement {
    const { logs } = this.props;
    const logsText = logs ? logs.join('\n') : '';
    const logsBlobUrl = getBlobUrl(logsText);

    return (
      <div className={styles.logsTools}>
        <Flex>
          <FlexItem>
            <a download="logs.txt" href={logsBlobUrl}>
              <DownloadIcon />
              Download
            </a>
          </FlexItem>
          <Divider isVertical />
          <FlexItem>
            <Button variant="link" onClick={() => this.handleExpand()}>
              {this.state.isExpanded ? (
                <React.Fragment>
                  <CompressIcon />
                  Compress
                </React.Fragment>
              ) : (
                <React.Fragment>
                  <ExpandIcon />
                  Expand
                </React.Fragment>
              )}
            </Button>
          </FlexItem>
        </Flex>
      </div>
    );
  }
}

export default LogsTools;
