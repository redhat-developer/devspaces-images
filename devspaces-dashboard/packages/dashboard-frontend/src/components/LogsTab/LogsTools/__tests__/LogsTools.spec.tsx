/*
 * Copyright (c) 2018-2021 Red Hat, Inc.
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *   Red Hat, Inc. - initial API and implementation
 */

import React from 'react';
import renderer, { ReactTestRenderer } from 'react-test-renderer';
import LogsTools from '..';

jest.mock('../../../../services/helpers/tools', () => {
  return {
    getBlobUrl: (text: string) => encodeURI(`https://fake.blob.url?blobText=${text}`),
  };
});

describe('The LogsTools component', () => {
  it('should render the component correctly', () => {
    const logs = [
      'Pulling image "quay.io/eclipse/che-theia-endpoint-runtime-binary:next"',
      'Successfully pulled image "quay.io/eclipse/che-theia-endpoint-runtime-binary:next"',
      'Created container remote-runtime-injectorvpj',
      'Started container remote-runtime-injectorvpj',
      'Pulling image "quay.io/eclipse/che-plugin-artifacts-broker:v3.4.0"',
      'Successfully pulled image "quay.io/eclipse/che-plugin-artifacts-broker:v3.4.0"',
      'Created container che-plugin-artifacts-broker-v3-4-0',
      'Started container che-plugin-artifacts-broker-v3-4-0',
      'Pulling image "quay.io/eclipse/che-jwtproxy:0.10.0"',
    ];
    const handleExpand = jest.fn();

    const component = renderComponent(logs, handleExpand);

    expect(handleExpand).not.toBeCalled();
    expect(component.toJSON()).toMatchSnapshot();
  });
});

function renderComponent(
  logs: string[],
  handleExpand: (isExpand: boolean) => void,
): ReactTestRenderer {
  return renderer.create(<LogsTools logs={logs} handleExpand={handleExpand} preventPostMessage />);
}
