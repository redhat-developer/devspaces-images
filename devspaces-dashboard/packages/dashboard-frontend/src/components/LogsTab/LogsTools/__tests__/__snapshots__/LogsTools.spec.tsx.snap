// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The LogsTools component should render the component correctly 1`] = `
<div>
  <div
    className="pf-l-flex"
  >
    <div
      className=""
    >
      <a
        download="logs.txt"
        href="https://fake.blob.url?blobText=Pulling%20image%20%22quay.io/eclipse/che-theia-endpoint-runtime-binary:next%22%0ASuccessfully%20pulled%20image%20%22quay.io/eclipse/che-theia-endpoint-runtime-binary:next%22%0ACreated%20container%20remote-runtime-injectorvpj%0AStarted%20container%20remote-runtime-injectorvpj%0APulling%20image%20%22quay.io/eclipse/che-plugin-artifacts-broker:v3.4.0%22%0ASuccessfully%20pulled%20image%20%22quay.io/eclipse/che-plugin-artifacts-broker:v3.4.0%22%0ACreated%20container%20che-plugin-artifacts-broker-v3-4-0%0AStarted%20container%20che-plugin-artifacts-broker-v3-4-0%0APulling%20image%20%22quay.io/eclipse/che-jwtproxy:0.10.0%22"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 512 512"
          width="1em"
        >
          <path
            d="M216 0h80c13.3 0 24 10.7 24 24v168h87.7c17.8 0 26.7 21.5 14.1 34.1L269.7 378.3c-7.5 7.5-19.8 7.5-27.3 0L90.1 226.1c-12.6-12.6-3.7-34.1 14.1-34.1H192V24c0-13.3 10.7-24 24-24zm296 376v112c0 13.3-10.7 24-24 24H24c-13.3 0-24-10.7-24-24V376c0-13.3 10.7-24 24-24h146.7l49 49c20.1 20.1 52.5 20.1 72.6 0l49-49H488c13.3 0 24 10.7 24 24zm-124 88c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20zm64 0c0-11-9-20-20-20s-20 9-20 20 9 20 20 20 20-9 20-20z"
          />
        </svg>
        Download
      </a>
    </div>
    <hr
      className="pf-c-divider pf-m-vertical"
    />
    <div
      className=""
    >
      <button
        aria-disabled={false}
        aria-label={null}
        className="pf-c-button pf-m-link"
        data-ouia-component-id="OUIA-Generated-Button-link-1"
        data-ouia-component-type="PF4/Button"
        data-ouia-safe={true}
        disabled={false}
        onClick={[Function]}
        role={null}
        type="button"
      >
        <svg
          aria-hidden={true}
          aria-labelledby={null}
          fill="currentColor"
          height="1em"
          role="img"
          style={
            Object {
              "verticalAlign": "-0.125em",
            }
          }
          viewBox="0 0 448 512"
          width="1em"
        >
          <path
            d="M0 180V56c0-13.3 10.7-24 24-24h124c6.6 0 12 5.4 12 12v40c0 6.6-5.4 12-12 12H64v84c0 6.6-5.4 12-12 12H12c-6.6 0-12-5.4-12-12zM288 44v40c0 6.6 5.4 12 12 12h84v84c0 6.6 5.4 12 12 12h40c6.6 0 12-5.4 12-12V56c0-13.3-10.7-24-24-24H300c-6.6 0-12 5.4-12 12zm148 276h-40c-6.6 0-12 5.4-12 12v84h-84c-6.6 0-12 5.4-12 12v40c0 6.6 5.4 12 12 12h124c13.3 0 24-10.7 24-24V332c0-6.6-5.4-12-12-12zM160 468v-40c0-6.6-5.4-12-12-12H64v-84c0-6.6-5.4-12-12-12H12c-6.6 0-12 5.4-12 12v124c0 13.3 10.7 24 24 24h124c6.6 0 12-5.4 12-12z"
          />
        </svg>
        Expand
      </button>
    </div>
  </div>
</div>
`;
